web/test/ShoppingCartTest.js

```diff
@@ -0,0 +1,57 @@
+/* global describe it before */
+const expect = require('chai').expect
+const Router = function () {
+  const history = []
+  const navigate = (location) => {
+    history.push(location)
+  }
+
+  return { navigate, history }
+}
+const productFactory = require('../scripts/products/Product.js').factory
+const ShoppingCartFactory = require('../scripts/cart/ShoppingCart').factory
+
+describe('ShoppingCart', function () {
+  const router = new Router()
+  const Product = productFactory(router)
```

Comment:
Is there a need to use router? I'm not sure if router is needed in the ShoppingCart since you might not need to use it.

Reply:
Good thinking. I tried removing router, and it still works. I guess the dependencies are optional if they are not being used.